waffle-ens/src/EnsBuilder.ts

```diff
@@ -0,0 +1,61 @@
+import {ENSRegistry, FIFSRegistrar} from '@ensdomains/ens';
+import {PublicResolver} from '@ensdomains/resolver';
+import {constants, Contract, utils, Wallet} from 'ethers';
+import {COIN_TYPE_ETH, deployContract} from './utils';
+
+const {namehash} = utils;
+const {HashZero} = constants;
+
+export class ENSBuilder {
+  wallet: Wallet;
+  ens?: Contract;
+  resolver?: Contract;
+  registrarMD?: Contract;
+  registrarSLD?: Contract;
+
+  constructor(wallet: Wallet) {
+    this.wallet = wallet;
+  }
+
+  async deployENSRegistry() {
+    return deployContract(this.wallet, ENSRegistry, []);
+  }
+
+  async deployENSResolver() {
+    return deployContract(this.wallet, PublicResolver, [this.ens!.address]);
+  }
+
+  async setupResolver() {
+    const resolverNode = namehash('resolver');
+    const resolverLabel = utils.id('resolver');
+    await this.ens!.setSubnodeOwner(HashZero, resolverLabel, this.wallet.address);
+    await this.ens!.setResolver(resolverNode, this.resolver!.address);
+    await this.resolver!.setAddr(resolverNode, COIN_TYPE_ETH, this.resolver!.address);
+  }
+
+  async deployENSRegistrar(namehash: string) {
+    return deployContract(this.wallet, FIFSRegistrar, [this.ens!.address, namehash]);
+  }
+
+  async setupRegistrarMD() {
+    const registrarLabel = utils.id('test');
+    await this.ens!.setSubnodeOwner(HashZero, registrarLabel, this.registrarMD!.address);
+  }
+
+  async createDomain(label: string) {
+    const domainLabel = utils.id(label);
+    await this.registrarMD!.register(domainLabel, this.wallet.address);
+    const domainNode = namehash(label + '.test');
+    await this.ens!.setResolver(domainNode, this.resolver!.address);
+    this.registrarSLD = await this.deployENSRegistrar(domainNode);
+    await this.ens!.setOwner(domainNode, this.registrarSLD.address);
+  }
+
+  async buildENS() {
+    this.ens = await this.deployENSRegistry();
+    this.resolver = await this.deployENSResolver();
+    await this.setupResolver();
+    this.registrarMD = await this.deployENSRegistrar(namehash('test'));
+    await this.setupRegistrarMD();
+  }
```

Comment:
#simplifyEnsBuilder

Deployment of `.test`, should not be a part of it anymore.